### PR TITLE
Fix summary handling in conftest.py

### PR DESCRIPTION
**Pull request by dmitryduev**
Description
-----------
This PR addresses a major source of flake in the refactored (`pytest`) unit tests.

TLDR:
Run summary may be updated multiple times during a test, but we are only interested in the last one.
We can have multiple runs saved to the relay server context, so we need to group by run id and take the last one for each run.

The `pandas` magic applied is the following:
```python
>>> import pandas as pd
>>> df = pd.DataFrame.from_records([{"_run": 1}, {"_run": 1, "a": 1}, {"_run": 2},{"_run": 2, "a": 2}])
>>> df
   _run    a
0     1  NaN
1     1  1.0
2     2  NaN
3     2  2.0
>>> df.groupby("_run").last().reset_index(level=["_run"])
   _run    a
0     1  1.0
1     2  2.0
```

Finally, run summary DataFrame must have only one row for the given run id, so in `Context.get_run_summary` we convert the summary to dict and extract the first (and only) row.

Testing
-------
How was this PR tested?

Checklist
-------
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

File: tests/unit_tests/conftest.py [hunk 1]
```diff
@@ -938,7 +938,17 @@ def summary(self) -> pd.DataFrame:
...
 
-        self._summary = self.get_file_contents("wandb-summary.json")
+        _summary = self.get_file_contents("wandb-summary.json")
+
+        # run summary may be updated multiple times,
+        # but we are only interested in the last one.
+        # we can have multiple runs saved to context,
+        # so we need to group by run id and take the
+        # last one for each run.
+        self._summary = (
```
Comment (kptkin) [inline tests/unit_tests/conftest.py:948]: Didn’t we say the problem is with merge? It doesn’t seem like this is solving this, unless it is expected to have multiple lines per run I’d…

**Review by kptkin — APPROVED** [review]
